mautic/mautic
# Removing options for TimeType that would cause an exception in Symfony 5.0

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [n]
| Deprecations?                          | [n]
| BC breaks? (use the c.x branch)        | [n]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-277



#### Description:

There is a note in the doc for upgrading apps from Symfony 4.4 to 5.0:

```
Removed support for using different values for the "model_timezone" and "view_timezone" options of the TimeType without configuring a reference date.
```

But when I add the [reference_date](https://symfony.com/doc/4.4/reference/forms/types/time.html#reference-date) option then it will prepend the time with the `1970-01-01` date. That is by design from [what is visible in the code](https://github.com/symfony/symfony/blob/5.0/src/Symfony/Component/Form/Extension/Core/Type/TimeType.php#L77-L87). So we cannot use that option. 

Instead I'm removing the 2 config params (`model_timezone`, `view_timezone`) that require the use of `reference_date` because it does not make sense to use them in this case. We are storing a time for a contact but we do not configure the timezone of the specific contact. We do not configure the timezone for the user loading the custom filed value either. We do not configure the timezone for the instance either. We configure timezone configured for the server(!) on which Mautic is running. This does not make any sense why we would convert time by the difference between the UTC and the timezone configured in the server. It won't help anyone. And the server timezone should be in UTC also.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a Time custom field
3. Save a time value to the custom field in a contact edit form. Let's say for 20:00.
4. Check that if you reload the page that the value stay the same.
5. Check that the value in the `leads` table is also the same.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11040"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
